### PR TITLE
feat: set defaults and clean config file

### PR DESCRIPTION
**Pull request by jooola**
- Clean the legacy config ingesting
- Add defaults to reduce the size of required fields in the config file
- ~~Replace url parts with a single public_url entry~~
-  Remove unused web server user config entry

### CODE REVIEW

**Comment by jooola** [comment]
I would like to have this in alpha.11

**Comment by jooola** [comment]
Maybe the change related to #1574 should not come here, They require to rework the web installer, and this is difficult to get right on the first time.

**Comment by jooola** [comment]
This is ready to be merged, I removed anything related to #1574.

Please de not squash

File: docs/admin-manual/setup/configuration.md [hunk 1]
```diff
@@ -3,176 +3,191 @@ title: Configuration
...
 
+```ini
 [general]
-api_key = XXXXXXXXXXXXXXXXXXXXX
-web_server_user = www-data
-airtime_dir = /usr/share/airtime
-base_url = libretime.example.com
+# The internal API authentication key. This field is required.
+; api_key =
+api_key = f985635ac757e05715bc2d2b1f7d7045
```
Comment (paddatrapper) [inline docs/admin-manual/setup/configuration.md:16]: Maybe set this to a clearly fake key? e.g. `0000000000000000000000000` or `0123456789abcdef`

File: docs/admin-manual/setup/configuration.md [hunk 2]
```diff
@@ -3,176 +3,191 @@ title: Configuration
...
+
+# The public url scheme. Mutually exclusive with force_ssl.
+; protocol =
+protocol =
```
Comment (paddatrapper) [inline docs/admin-manual/setup/configuration.md:20]: You've got the same thing twice, one commented and one not. What is the reason for the duplication?

File: docs/admin-manual/setup/configuration.md [hunk 3]
```diff
@@ -3,176 +3,191 @@ title: Configuration
...
+# The public url hostname.
+; base_url = localhost
+base_url = localhost
+# The public url port.
+; base_port = localhost
```
Comment (paddatrapper) [inline docs/admin-manual/setup/configuration.md:25]:
```suggestion
; base_port = 80
```

File: docs/admin-manual/setup/configuration.md [hunk 4]
```diff
@@ -3,176 +3,191 @@ title: Configuration
...
+port = 5432
+# The name of the PostgreSQL database.
+; name = libretime
+name = airtime
+# The username of the PostgreSQL user.
+; user = libretime
+user = airtime
+# The password of the PostgreSQL user.
+; password = libretime
+password = f0611ea4849b028c0c7e34f0b4455c22
```
Comment (paddatrapper) [inline docs/admin-manual/setup/configuration.md:82]: Set this to an obviously fake value e.g. `libretime`

File: docs/admin-manual/setup/configuration.md [hunk 5]
```diff
@@ -3,176 +3,191 @@ title: Configuration
...
+port = 5672
+# The virtual host of RabbitMQ server.
+; vhost = /libretime
+vhost = /airtime
+# The username of the RabbitMQ user.
+; user = libretime
+user = airtime
+# The password of the RabbitMQ user.
+; password = libretime
+password = 2db48cc70ee197725f65c66ed27e0f95
```
Comment (paddatrapper) [inline docs/admin-manual/setup/configuration.md:119]:
```suggestion
password = libretime
```

File: legacy/tests/application/helpers/AirtimeInstall.php [hunk 6]
```diff
@@ -145,38 +145,38 @@ public static function DbConnect($p_exitOnError = true)
...
-
-                exit(1);
-            }
-            echo "* Giving Apache permission to access {$rp}" . PHP_EOL;
-            $success = chown($rp, $CC_CONFIG['webServerUser']);
-            $success = chgrp($rp, $CC_CONFIG['webServerUser']);
-            $success = chmod($rp, 0775);
-        }
-    }
+    // public static function InstallStorageDirectory()
```
Comment (paddatrapper) [inline legacy/tests/application/helpers/AirtimeInstall.php:148]: If you're commenting out this functionality, remove it. We have access to it from git anyway

File: docs/admin-manual/setup/configuration.md (hunk 2 above)
Comment (jooola) [reply docs/admin-manual/setup/configuration.md:20]: It was used to display the default values, but I already have a more explicit way to define default lying in some other branch, moving it in this one.

**Comment by jooola** [comment]
Applied the suggestions

**Comment by jooola** [comment]
I will also remove the deprecated fields from the example file.

**Comment by jooola** [comment]
Done

**Pull request merged**